<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"><head><title>Error (pyre-ast.PyreAst.Parser.Error)</title><link rel="stylesheet" href="../../../../odoc.css"/><meta charset="utf-8"/><meta name="generator" content="odoc 2.1.1"/><meta name="viewport" content="width=device-width,initial-scale=1.0"/><script src="../../../../highlight.pack.js"></script><script>hljs.initHighlightingOnLoad();</script></head><body class="odoc"><nav class="odoc-nav"><a href="../index.html">Up</a> â€“ <a href="../../../index.html">pyre-ast</a> &#x00BB; <a href="../../index.html">PyreAst</a> &#x00BB; <a href="../index.html">Parser</a> &#x00BB; Error</nav><header class="odoc-preamble"><h1>Module <code><span>Parser.Error</span></code></h1><p>This module contains a type that represents parsing errors.</p></header><div class="odoc-content"><div class="odoc-spec"><div class="spec type" id="type-t" class="anchored"><a href="#type-t" class="anchor"></a><code><span><span class="keyword">type</span> t</span><span> = </span><span>{</span></code><table><tr id="type-t.message" class="anchored"><td class="def record field"><a href="#type-t.message" class="anchor"></a><code><span>message : string;</span></code></td></tr><tr id="type-t.line" class="anchored"><td class="def record field"><a href="#type-t.line" class="anchor"></a><code><span>line : int;</span></code></td></tr><tr id="type-t.column" class="anchored"><td class="def record field"><a href="#type-t.column" class="anchor"></a><code><span>column : int;</span></code></td></tr><tr id="type-t.end_line" class="anchored"><td class="def record field"><a href="#type-t.end_line" class="anchor"></a><code><span>end_line : int;</span></code></td></tr><tr id="type-t.end_column" class="anchored"><td class="def record field"><a href="#type-t.end_column" class="anchor"></a><code><span>end_column : int;</span></code></td></tr></table><code><span>}</span></code></div><div class="spec-doc"><p>Line numbers start from 1 and column numbers start from 1.</p><p>Note that it is possible for any of these line/column numbers to be set to -1. This can happen either when CPython runtime encounters a non-syntax error, or when an unexpected internal error occurs during parsing. It is also possible that CPython runtime sometimes set the end locations to come before the start locations. If needed, clients are expected to perform their own validations on the location numbers -- this library is only responsible of forwarding whatever data that comes from the CPython side.</p></div></div></div></body></html>